### PR TITLE
Feature uri scheme fixes

### PR DESCRIPTION
**Pull request by Sunjammer**
This pull request implements the following fixes to the WebView uri scheme handling:

Ensure schemes read from the uno project are suffixed with :// before being taken into account by webviews. This might be seen as redundant so is in own patch.
Ensure schemes are only matched against the URL prefix: In the past the scheme "foo" would match on foo anywhere in the url.
Ensure no attempts to match schemes takes place if there is no explicitly set URI scheme handler for the webview.

This fixes #3859

This PR contains:
- [x] Changelog

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -25,6 +25,8 @@
 - Added `Reuse` to `Each` allowing the reuse of nodes
 - Added `OnChildMoved` to `Visual`. Anything implementing `OnChildAdded` or `OnChildRemoved` will likely need to implement `OnChildMoved` as well. This happens when a child's position in `Children` list changes.
 - Added `OnChildMovedWhileRooted` to `IParentObserver`
+### WebView
```
Comment (kusma) [inline CHANGELOG.md]: please add a space before this heading.

Comment (Sunjammer) [inline CHANGELOG.md]: damn that was fast

File: Source/Fuse.Controls.WebView/Android/FuseWebViewClient.java [hunk 2]
```diff
@@ -16,30 +17,27 @@
...
 	{
-		for(String uri : _customURIs)
-		{
-			if(url.indexOf(uri) != -1)
-			{
-				_onCustomURI.run(url);
-				return true;
-			}
-		}
+		if(tryInterceptUriScheme(url))
```
Comment (mortoray) [inline Source/Fuse.Controls.WebView/Android/FuseWebViewClient.java]: Might it not be cleaner to make the function `matchedInterceptUriScheme` and do the call to `_onCustomURI.run` here?

File: Source/Fuse.Controls.WebView/Android/WebView.uno [hunk 3]
```diff
@@ -158,9 +172,19 @@ namespace Fuse.Android.Controls
 		public void LoadUrl(string url)
 		{
 			if (url == null || url== "") url = "about:blank";
+			//This extra check is needed on android since setting the Url directly doesn't trigger shouldOverrideUrlLoading
+			if(HasURISchemeHandler())
+			{
+				foreach(string uri in _uriSchemes)
```
Comment (mortoray) [inline Source/Fuse.Controls.WebView/Android/WebView.uno]: This appears to be a duplication of the code in `FuseWebViewClient.java`.

**Comment by nekronos** [comment]
@Sunjammer bump

**Comment by Sunjammer** [comment]
Oh ffs I was convinced this thing got merged. I'll get on it.

**Comment by kusma** [comment]
@Sunjammer: The diff now is pretty much bonkers. Could you rebase this on master rather than doing these sketchy merges?

**Comment by kusma** [comment]
Basically, it's not this branch that is supposed to merge release-1.2 into master.

**Comment by Sunjammer** [comment]
For sure. TBH this merge was screwed repeatedly anyway due to me messing up which remote I was working off of. I'll clean her up.

**Comment by Sunjammer** [comment]
Re the last merge commit, should I not be using GitHub's conflict resolver to fix changelog conflicts?

**Comment by kusma** [comment]
@Sunjammer: I don't see a problem with doing that.

**Comment by Sunjammer** [comment]
@mortoray Look better now?

File: CHANGELOG.md [hunk 4]
```diff
@@ -54,6 +57,10 @@
 - Changed/fixed `Trigger` to provide the trigger itself as the target node to a `TriggerAction`, previously it'd provide the parent of the trigger. The old behaviour was due to an old tree structure. This should have been updated a long time ago. This allows actions to reference the trigger in which they are contained. If you've created a custom Uno `TriggerAction` and need the old behaviour modify your `Perform(Node target)` to use `target.Parent`. Triggers should in general scan upwards from the target node.
 
 ## Optimizations
+
+# 1.2
+
+=======
```
Comment (kusma) [inline CHANGELOG.md]: Please refrain from changing other parts of the changeling...

File: CHANGELOG.md [hunk 5]
```diff
@@ -75,7 +82,6 @@
 ### Fuse.Elements
 - Fixed an issue where the ElementBatcher ended up throwing an Exception complaining about the element not having a caching rect.
 
-
```
Comment (kusma) [inline CHANGELOG.md]: Same as above, please don't do unrelated changes in changeling.

File: Source/Fuse.Controls.WebView/Android/FuseWebViewClient.java [hunk 6]
```diff
@@ -48,12 +42,12 @@ public boolean shouldOverrideUrlLoading(WebView view, String url)
...
 		loadingFinished = false;
 		view.loadUrl(url);
-		return true;
+		return false;
 	}
 	
 	@Override
 	public void onPageStarted(WebView view, String url, Bitmap favIcon)
-	{
+	{
```
Comment (kusma) [inline Source/Fuse.Controls.WebView/Android/FuseWebViewClient.java]: Please don't.

**Comment by Sunjammer** [comment]
All I really did (I thought) was remove the merge markers so GitHub would let me mark the conflict as resolved. I absolutely effing loathe changelog merges, can't seem to do them right ever.

**Comment by kusma** [comment]
@Sunjammer: Seems you did more; you left at least one marker, and somehow deleted a line. Merging changelogs isn't my favourite thing either, but we still need to get it right.

**Comment by Sunjammer** [comment]
Once more unto the breach
